src/async/iterableUtils.ts

```diff
@@ -47,3 +47,79 @@ export function createReadableStream<TValue = unknown>() {
 
 	return [stream, controller] as const;
 }
+
+export function createServerEvent<
+	const TData,
+	const TEvent extends string,
+	const TId extends string,
+	const TRetry extends number,
+>(opts: {
```

Comment:
cooool

Reply:
In response to your question about abstractions for tRPC, I wrote some late night thoughts down (edited for clarity:

        I don't like useSubscription() for this. It feels like that abstraction
        should be reserved for websockets, since it's a totally different
        protocol. Perhaps some combination of the 'batchStreamLink' abstraction
        and a useStream() hook would be better?

		Or, maybe we can just overload useQuery, and stuff any more robust features like pagination & resumability into useInfiniteQuery

	-	We can send any event with any message, just like any EE. How can this simplify the api or implementation?
    -   What would be a 'sane default' for disconnect logic? error logic? retry logic?

- [ ] CONFIG CALLBACKS - expose to the user, much like the `responseMeta` callback
    - const calculateRetryDelay = (id, event, message) => number
    - const generateId = (event, message) => string
    - @OR is this overengineerinFFg? Maybe we just use the nonce or index or something?
    - @OR @OR provide the option to configure, and a sane default? <-- my fav-[

Is returning 200 semantically sane, or should we return 207 multi-status and stuff more metadata into the chunks to parse back individual status' on the client? For a single query without batching, 200 would be fine, but anything more complex and we lose information by relying on just the status code.